javascript/javascript1/week3/smart phone uasge app/smartphone.js

```diff
@@ -0,0 +1,61 @@
+let activities = [];
+
+function addActivity(activity, duration) {
+  if (typeof duration !== "number") {
+    console.log("Duration must be a number");
+  }
+  const today = new Date();
+  const date = today.toISOString().split("T")[0];
+  const activityRecord = {
+    date: date,
+    activity: activity,
+    duration: duration,
+  };
+  activities.push(activityRecord);
+}
+
+addActivity("Youtube", 30);
+addActivity("WhatsApp", 25);
+addActivity("Instagaram", 20);
+addActivity("WhatsApp", 15);
+
+console.log(activities);
+
+function showStatus() {
+  let usageLimit = 120;
+  if (activities.length === 0) {
+    console.log("Add some activities before calling showStatus");
+  }
+  const totalActivities = activities.length;
+  let totalDuration = 0;
+  for (let i = 0; i < totalActivities; i++) {
+    totalDuration += activities[i].duration;
+  }
+  if (totalDuration > usageLimit) {
+    console.log(
+      `You have reached your usage limit ${usageLimit} min., so no more smartphone usage`
+    );
+  }
+  return `You have added ${totalActivities} activites. They amount to ${totalDuration} min. of usage`;
+}
+
+console.log(showStatus());
+
+addActivity("LinkedIn", 35);
+
+console.log(showStatus());
+console.log(activities);
+
+function getMostTimeSpentActivity() {
```

Comment:
Excellent and cleanly formatted implementation and I like the feature you decided to add! 🤩 

If I were to run `addActivity("Youtube", 10)` after adding LinkedIn above. Would you still expect LinkedIn to be returned by this method? And why? If a change is needed, what would you do? 😄 (Mainly a thought exercise, so you do not have to implement unless you want to, of course.)

Reply:
I have done few changes to the code which still shows LinkedIn as my mostTimeSpentActivity, after adding Youtube, 10.

Reply:
I don't think introducing another global variable here is a good idea as it just opens us up to getting the data out of sync; your old approach did a great job about this! 😄  What I was fishing after is that only a single activity is being considered, but if I logged the same activity multiple times perhaps the sum should be considered instead? E.g., if I logged 30 minutes on YouTube on Monday and then another 10 on Tuesday, then YouTube should probably win over LinkedIn as it has 40 total minutes compared to LinkedIn's 35 (in the old code).

I hope this makes any sense.